src/provider/provider_devdax_memory.c

```diff
@@ -0,0 +1,647 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+#include <errno.h>
+#include <limits.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "base_alloc_global.h"
+#include "critnib.h"
+#include "provider_devdax_memory_internal.h"
+#include "provider_os_memory_internal.h"
+#include "utils_common.h"
+#include "utils_concurrency.h"
+#include "utils_log.h"
+
+#include <umf.h>
+#include <umf/memory_provider_ops.h>
+#include <umf/providers/provider_devdax_memory.h>
+
+#define NODESET_STR_BUF_LEN 1024
+
+#define TLS_MSG_BUF_LEN 1024
+
+typedef struct os_last_native_error_t {
+    int32_t native_error;
+    int errno_value;
+    char msg_buff[TLS_MSG_BUF_LEN];
+} os_last_native_error_t;
+
+static __TLS os_last_native_error_t TLS_last_native_error;
+
+// helper values used only in the Native_error_str array
+#define _UMF_DEVDAX_RESULT_SUCCESS                                             \
+    (UMF_DEVDAX_RESULT_SUCCESS - UMF_DEVDAX_RESULT_SUCCESS)
+#define _UMF_DEVDAX_RESULT_ERROR_ALLOC_FAILED                                  \
+    (UMF_DEVDAX_RESULT_ERROR_ALLOC_FAILED - UMF_DEVDAX_RESULT_SUCCESS)
+#define _UMF_DEVDAX_RESULT_ERROR_ADDRESS_NOT_ALIGNED                           \
+    (UMF_DEVDAX_RESULT_ERROR_ADDRESS_NOT_ALIGNED - UMF_DEVDAX_RESULT_SUCCESS)
+#define _UMF_DEVDAX_RESULT_ERROR_FREE_FAILED                                   \
+    (UMF_DEVDAX_RESULT_ERROR_FREE_FAILED - UMF_DEVDAX_RESULT_SUCCESS)
+#define _UMF_DEVDAX_RESULT_ERROR_PURGE_FORCE_FAILED                            \
+    (UMF_DEVDAX_RESULT_ERROR_PURGE_FORCE_FAILED - UMF_DEVDAX_RESULT_SUCCESS)
+
+static const char *Native_error_str[] = {
+    [_UMF_DEVDAX_RESULT_SUCCESS] = "success",
+    [_UMF_DEVDAX_RESULT_ERROR_ALLOC_FAILED] = "memory allocation failed",
+    [_UMF_DEVDAX_RESULT_ERROR_ADDRESS_NOT_ALIGNED] =
+        "allocated address is not aligned",
+    [_UMF_DEVDAX_RESULT_ERROR_FREE_FAILED] = "memory deallocation failed",
+    [_UMF_DEVDAX_RESULT_ERROR_PURGE_FORCE_FAILED] = "force purging failed",
+};
+
+static void os_store_last_native_error(int32_t native_error, int errno_value) {
```

Comment:
`os_`? either `devdax_` or drop the prefix (I can see in L0 prov. we have just `store_last_native_error`)

pls apply in other functions/vars in here, as well

Reply:
Done